boards/nrf52dk/src/main.rs

```diff
@@ -153,41 +153,50 @@ pub unsafe fn reset_handler() {
     uicr.set_psel1_reset_pin(BUTTON_RST_PIN);
 
     // GPIOs
-    // FIXME: Test if it works and remove un-commented code!
     let gpio_pins = static_init!(
-        [&'static nrf5x::gpio::GPIOPin; 15],
-        [&nrf5x::gpio::PORT[3],  // Bottom right header on DK board
-        &nrf5x::gpio::PORT[4],   //
-        &nrf5x::gpio::PORT[28],  //
-        &nrf5x::gpio::PORT[29],  //
-        &nrf5x::gpio::PORT[30],  //
-        &nrf5x::gpio::PORT[31],  // -----
-        &nrf5x::gpio::PORT[10],  // Top right header on DK board
-        &nrf5x::gpio::PORT[9],   //
-        &nrf5x::gpio::PORT[8],   //
-        &nrf5x::gpio::PORT[7],   //
-        &nrf5x::gpio::PORT[6],   //
-        &nrf5x::gpio::PORT[5],   //
-        &nrf5x::gpio::PORT[21],  //
-        &nrf5x::gpio::PORT[1],   //
-        &nrf5x::gpio::PORT[0],   // -----
-        /*&nrf52::gpio::PORT[18],  // Top mid header on DK board
-        &nrf52::gpio::PORT[17],  //
-        &nrf52::gpio::PORT[16],  //
-        &nrf52::gpio::PORT[15],  //
-        &nrf52::gpio::PORT[14],  //
-        &nrf52::gpio::PORT[13],  //
-        &nrf52::gpio::PORT[12],  //
-        &nrf52::gpio::PORT[11],  // ----
-        &nrf52::gpio::PORT[27],  // Top left header on DK board
-        &nrf52::gpio::PORT[26],  //
-        &nrf52::gpio::PORT[2],  //
-        &nrf52::gpio::PORT[25],  //
-        &nrf52::gpio::PORT[24],  //
-        &nrf52::gpio::PORT[23],  //
-        &nrf52::gpio::PORT[22],  //
-        &nrf52::gpio::PORT[20],  //
-        &nrf52::gpio::PORT[19],  // ----*/
+        [&'static nrf5x::gpio::GPIOPin; 23],
+        [
+        // Bottom right header on DK board
+        &nrf5x::gpio::PORT[3],
+        &nrf5x::gpio::PORT[4],
+        &nrf5x::gpio::PORT[28],
+        &nrf5x::gpio::PORT[29],
+        &nrf5x::gpio::PORT[30],
+        &nrf5x::gpio::PORT[31],
+        // -----
+        // Top right header on DK board
+        // NOTE: Those are connected internally and should not be used externally
```

Comment:
Right, therefore I think we don't want to expose them

Reply:
That's fine for me. I assume, we must then ensure that there are enough capsule drivers that provide access to all the devices connected to those pins, right?

Reply:
Yeah.

The RX/TX/RTS/CTS lines are controlled by the UART driver

The XL1, XL2 lines are the crystal, which aren't exposed by a driver, but of course are used by the hardware. I suspect the pins are exposed on the board to monitor it with a scope or something.

We don't have a driver for NFC. It would be great to build one, though we would somehow need access to the NFC specification (which as far as I can tell is under NDA + a paywall)

Reply:
Oh, but I see these were already enabled in `master`... oy...

@Vagabond and @JayKickliter are the known users of NRF52DK for actual applications. Maybe they have some insight as to (a) whether changes to the exposed pins would break existing applications and (b) what the "right" set of pins to expose would be.

Reply:
Thanks for your reply. Maybe I should wait for the user's responses before commenting-out those lines?